generator/src/test/java/com/graphicsfuzz/generator/semanticschanging/InterchangeExprMutationFinderTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 The GraphicsFuzz Project Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.graphicsfuzz.generator.semanticschanging;
+
+import com.graphicsfuzz.common.transformreduce.ShaderJob;
+import com.graphicsfuzz.common.util.IRandom;
+import com.graphicsfuzz.common.util.RandomWrapper;
+import com.graphicsfuzz.common.util.ShaderJobFileOperations;
+import com.graphicsfuzz.generator.mutateapi.Mutation;
+import com.graphicsfuzz.util.ToolPaths;
+import java.io.File;
+import java.nio.file.Paths;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.junit.Assert.*;
+
+public class InterchangeExprMutationFinderTest {
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Test
+  public void testValidInterchanges() throws Exception {
+    final IRandom generator = new RandomWrapper(0);
+    final ShaderJobFileOperations fileOperations = new ShaderJobFileOperations();
+    for (File shaderJobFile : Paths.get(ToolPaths.getShadersDirectory(), "samples",
```

Comment:
Technically there is `listShaderJobFiles` for this. Not sure how important this is though. It ensures they are sorted and so makes things more deterministic.

Reply:
Done, thanks!